packages/devtools_app/lib/src/flutter/banner_messages.dart

```diff
@@ -68,62 +101,76 @@ class _BannerMessageContainerState extends State<BannerMessageContainer>
     final newController = Controllers.of(context)?.bannerMessages;
     if (newController == controller) return;
     controller = newController;
+  }
+
+  void push(BannerMessage message) {
+    if (controller.isMessageDismissed(message) ||
```

Comment:
this logic should be in the controller instead of here. nobody should ever accidentally add a dismissed message or an already visible message to the controller.